proxy/proxy2.py

```diff
@@ -366,14 +378,35 @@ def parse_qsl(s):
             if res_body_text:
                 print with_color(32, "==== RESPONSE BODY ====\n%s\n" % res_body_text)
 
+    def intercept_handler(self, req):
+        matched_entries = []
+        # match url
+        database_objs = RewriteRules.objects.values('url', 'headers', 'response')
+        for regex in database_objs:
+            if re.match(regex['url'], req.path):
+                matched_entries.append(regex)
+        headers = {h.lower().replace('-', '_'): req.headers[h] for h in req.headers}
+        if not matched_entries:
+            return
+        # match headers
+        for entry in matched_entries:
+            for header in entry['headers']:
+                lowercase = header.lower().replace('-', '_')
+                if lowercase in headers and re.match(entry['headers'][header], headers[lowercase]):
+                    continue
+                break
```

Comment:
If we break out some of this loop into another method, it would help increase readability and a comment or two will go a long way here.